### PR TITLE
Customization

### PR DESCRIPTION
**Pull request by lbartoletti**
This PR adds the ability to use customization.

The test with travis is an example for that.

### CODE REVIEW

**Comment by elemoine** [comment]
Thanks @lbartoletti. I'll have a look later today.

File: .build/travis_script.sh [hunk 1]
```diff
@@ -76,4 +76,15 @@ printf "travis_fold:start:upgrade\nRun upgrade"
 pum upgrade -p qwat_prod -t qwat_sys.info -d $DELTA_DIRS
 echo "travis_fold:end:upgrade"
 
+# Extend qwat_prod with a customization
+printf "travis_fold:start:init-sire\nExtend database with ch_vd_sire"
```
Comment (elemoine) [inline .build/travis_script.sh]: ch_vd_sire ?

File: .build/travis_script.sh [hunk 2]
```diff
@@ -76,4 +76,15 @@ printf "travis_fold:start:upgrade\nRun upgrade"
...
+$TRAVIS_BUILD_DIR/customizations/sigip/init.sh -p qwat_prod -s 21781
```
Comment (elemoine) [inline .build/travis_script.sh]: I'd name the file `init_sigip.sh` for consistency with `init_qwat.sh` and `init_sire.sh`.

File: customizations/sigip/damage/damage.sql [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  label_2_rotation double precision,
+  label_2_text character varying(120),
+  CONSTRAINT damage_pkey PRIMARY KEY (id)
+);
+
+GRANT SELECT, REFERENCES, TRIGGER ON TABLE qwat_od.damage TO qwat_viewer;
+GRANT ALL ON TABLE qwat_od.damage TO qwat_user;
+GRANT ALL ON TABLE qwat_od.damage TO qwat_manager;
+COMMENT ON TABLE qwat_od.damage
+IS 'pipe damage';
```
Comment (elemoine) [inline customizations/sigip/damage/damage.sql]: Is this customization added only for the purpose of testing customization on Travis? I find it strange to add a `customizations` directory to the qwat repo while customizations are, by definition, outside the qwat repo.

File: update/upgrade_db.sh [hunk 4]
```diff
@@ -7,14 +7,22 @@
...
+
 # Default values
 SRID=21781
 CLEAN=0
 LOCALDIRGIVEN=0
-LOCALDIR=/home/regis/OSLANDIA/projets_locaux/QWAT/local_update_dir_test/
 TMPFILEDUMP=/tmp/qwat_dump
 UPGRADE=0
 SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+#LOCALDIR="$( cd "$( "${SCRIPTDIR}/../customizations/sigip" )" && pwd)"
```
Comment (elemoine) [inline update/upgrade_db.sh]: Do you really want to keep that comment line?

File: update/upgrade_db.sh [hunk 5]
```diff
@@ -28,25 +36,25 @@ case $key in
     -h|--help)
         echo "Arguments:"
         echo -e "\t-c|--clean\t\tCleans comp and test DB before starting"
-        # echo -e "\t-d|--localdirpath\tRuns a second update cycle with local customization delta files"
+        echo -e "\t-d|--localdirpath\tRuns a second update cycle with local customization delta files"
```
Comment (elemoine) [inline update/upgrade_db.sh]: In my opinion there should not be a second update cycle. The core, extensions and customizations should be updated as part of the same pum upgrade cycle.

File: customizations/sigip/damage/damage.sql (hunk 3 above)
Comment (lbartoletti) [reply customizations/sigip/damage/damage.sql]: Yes, it's only for Travis and set an example.

File: .build/travis_script.sh (hunk 2 above)
Comment (lbartoletti) [reply .build/travis_script.sh]: Right.

File: .build/travis_script.sh (hunk 1 above)
Comment (lbartoletti) [reply .build/travis_script.sh]: Oops, copy/paste

File: customizations/sigip/damage/damage.sql (hunk 3 above)
Comment (elemoine) [reply customizations/sigip/damage/damage.sql]: I'd suggest that we move the customizations directory to .build.

**Comment by lbartoletti** [comment]
Thanks @elemoine

File: update/upgrade_db.sh [hunk 6]
```diff
@@ -28,25 +35,25 @@ case $key in
     -h|--help)
         echo "Arguments:"
         echo -e "\t-c|--clean\t\tCleans comp and test DB before starting"
-        # echo -e "\t-d|--localdirpath\tRuns a second update cycle with local customization delta files"
+        echo -e "\t-d|--localdirpath\tAdds local customization delta files"
```
Comment (elemoine) [inline update/upgrade_db.sh]: What if I use both an extension and a customization? Will I be able to use upgrade_db.sh? I understand that upgrade_db.sh supports only one additional extension/customization, but I may be wrong.

**Comment by lbartoletti** [comment]
release 1.3.3 #250

**Pull request merged**